### PR TITLE
Spit log message whenever we retry processing an event

### PR DESCRIPTION
**Pull request by mjward**
## Context

The default behaviour when an error occurs for EventProcessors in EventSourcery is to kill the process and immediately respawn and try again. 

@jiexinhuang created the `ExponentialBackoffRetry` error handler for RSS, which wouldn't kill the process, it would as the class name indicates, exponentially back off retrying processing the failed event until successful. 

The issue we have with this is there is not a lot of transparency as to how often this is occurring, and/or how long it takes to be successful upon retry.   

## Change

An additional line in the logs should highlight how often retries are occurring together with how long it takes for that retry to be successful. 

## Considerations

This only impacts you if you are using the `ExponentialBackoffRetry` error handler (which I think might only be RSS? not 100% on that)

Paired with @charithperera

### CODE REVIEW

File: lib/event_sourcery/event_processing/error_handlers/exponential_backoff_retry.rb [hunk 1]
```diff
@@ -26,6 +26,7 @@ def with_error_handling
           report_error(error)
           update_retry_interval(error)
           sleep(@retry_interval)
+          EventSourcery.logger.info { "Retrying #{@processor_name} with error: #{error.message} at interval=#{@retry_interval}" }
```
Comment (jiexinhuang) [inline lib/event_sourcery/event_processing/error_handlers/exponential_backoff_retry.rb]: Perhaps it should log at `error` level instead of `info`?

**Comment by amanda08** [comment]
:+1: This would be super useful. We're using `ExponentialBackoffRetry` in Shoebox as well.

File: lib/event_sourcery/event_processing/error_handlers/exponential_backoff_retry.rb (hunk 1 above)
Comment (mjward) [reply lib/event_sourcery/event_processing/error_handlers/exponential_backoff_retry.rb]: Its not really an error though, the error was the thing that caused the retry in the first place. This is more just info that indicates that its going to retry processing.

**Comment by mjward** [comment]
@twe4ked just looked at the CHANGELOG and it looks like you have had stuff there pending a release. Seeing as this project is now open source, removing functionality which is potentially a breaking change feels like a major release? Would you agree?

**Comment by twe4ked** [comment]
Yeah that's what made me think about it. I think a `0.16.0` is probably a good idea. We'll need to release the `-postgres` gem too because it relies on one of the removed methods IIRC. Sorry.

**Comment by mjward** [comment]
According to https://semver.org which I like to adhere to anything that breaks the API in an incompatible manner would be a major release. Hence we are looking at `1.0`

Removing core method that drive event processors feels like a heavy change to me (even if we haven't personally used them at Envato for awhile). Open sourcing this repo requires us to be predictable and consistent with versioning otherwise we run the risk of breaking other peoples projects that may be now relying on this gem.

**Comment by orien** [comment]
Before version 1.0 is released, any change is allowed. I don't think we should release a version 1.0 until we ready to commit to the API.

https://semver.org/#spec-item-4

> 4. Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

**Comment by joesustaric** [comment]
@mjward and @twe4ked I had a similar brief discussion RE: versioning for the msql version of this gem (yet to be open sauced).. 
< v 1 is predictable .. that is it should be treated as a rapidly changing unstable API to anyone using it. bumping the minor version when its < v1 users should not be surprised that there might be something breaking..

I think we should have been on v1 already with this gem tbh.. feels like it's been 'stable' for a while now... +1 for bumping it to v1.0  feels like its ready regardless. :)

**Comment by joesustaric** [comment]
haha @orien we posted almost at the same time.. But yeah I see your point.. we should agree on the 'stable' API then v1 it. 👍

**Comment by mjward** [comment]
okie dokes, happy to keep it at `0.16` just want to make sure we are all on the same page. I do think its probably overdue for a review of a bump to `1.0` though as @joesustaric mentions. It can't stay in development forever.

**Comment by orien** [comment]
To be clear I'm not against releasing a `1.0` now. Rather that it should be a conscious choice to do so.

And to clarify, Semantic Versioning isn't forcing us to do so with this change.

**Comment by joesustaric** [comment]
Look to the semantic versioning for guidance 🔮 
>How do I know when to release 1.0.0?
If your software is being used in production, it should probably already be 1.0.0. If you have a stable API on which users have come to depend, you should be 1.0.0. If you’re worrying a lot about backwards compatibility, you should probably already be 1.0.0.

**Pull request merged**